Source/adventureDAO.js

```diff
@@ -383,29 +377,37 @@ exports.endRound = async function (adventure, thread) {
 			})
 
 			// Replace placeholder
-			adventure.room.moves.splice(adventure.room.moves.findIndex(move => move.userReference.team === "enemy" && move.userReference.index == index), 1);
-			if (counterpartHasPriority) {
-				adventure.room.priorityMoves.push(move);
-			} else {
-				adventure.room.moves.push(move);
+			let placeholderIdx=adventure.room.moves.findIndex(move => move.userReference.team === "enemy" && move.userReference.index == index)
+			if(placeholderIdx >=0){
+				adventure.room.moves.splice(placeholderIdx, 1,move);
 			}
 		}
 	});
 
 
 	// Randomize speed ties
-	[adventure.room.priorityMoves, adventure.room.moves].forEach(moveQueue => {
-		moveQueue.forEach(move => {
-			move.speed += generateRandomNumber(adventure, 10, "battle") / 10;
-		})
-		moveQueue.sort((first, second) => {
-			return second.speed - first.speed;
-		})
+	let randomOrderBag=Array(adventure.room.moves.length).fill().map((_,idx)=>idx) // ensure that unique values are available for each move
+	adventure.room.moves.forEach(move => {
+		let rIdx = generateRandomNumber(adventure, randomOrderBag.length, "battle");
+		move.randomOrder = randomOrderBag.splice(rIdx,1)[0]; // pull a remaining randomOrder out of the bag and assign it to a move
+	})
+	// "second" is the 1st parameter rather than "first", so that way the sort is descending instead of ascending
+	adventure.room.moves.sort((second, first) => {
+		if(first.priority === second.priority && first.speed === second.speed && first.randomOrder === second.randomOrder){
+			return 0;
+		}
+		else if((first.priority > second.priority) || 
+		(first.priority === second.priority && first.speed > second.speed) ||
+		(first.priority === second.priority && first.speed === second.speed && first.randomOrder > second.randomOrder))
+		{
+			return 1;
+		}
+		else return -1;
```

Comment:
```suggestion
		if (first.priority > second.priority) {
			return 1;
		} else if (first.priority < second.priority) {
			return -1;
		} else {
		if (first.speed > second.speed) {
		  return 1;
  		  } else if (first.speed < second.speed) {
  		    return -1;
  		  } else {
  		  if (first.randomOrder > second.randomOrder) {
  		  return 1;
  		  } else if (first.randomOrder < second.randomOrder) {
  		  return -1;
  		  } else {
  		  return 0;
  		  }
		}
```
I don't *think* this is hugely less performant than the above, and it's how I do the sorting in my head (ie I'm guessing I'll find it more readable in the future).

Reply:
```suggestion
if (first.priority == second.priority) {
  if (first.speed == second.speed) {
    return first.randomOrder - second.randomOrder;
  } else {
    return first.speed - second.speed;
  }
} else {
  return first.priority - second.priority;
}
```
Made a minimized line version for fun.

Reply:
this one... speaks pretty clearly actually.
I had the original way because that's how I've been shown to do tuple comparisons. (a chain of OR'd AND's of comparisons and tiebreakers.

Reply:
this way shows that IF you find stuff equal, you THEN zoom in for the tiebreaks